### PR TITLE
Improve chunk generation performance

### PR DESCRIPTION
**Pull request by Seb-stian**

### CODE REVIEW

File: Obsidian.API/Noise/TerrainSelect.cs [hunk 1]
```diff
@@ -4,20 +4,21 @@ namespace Obsidian.API.Noise;
...
-    internal Dictionary<Biomes, Module> TerrainModules { get; set; } = new();
+    private static readonly Constant zero = new() { ConstantValue = 0 };
 
     public Module BiomeSelector { get; set; }
+    internal Dictionary<Biomes, Module> TerrainModules { get; set; } = new();
 
     public TerrainSelect(Module biomeSelector)
     {
-        Source1 = new Constant { ConstantValue = 0 };
+        Source1 = zero;
```
Comment (Jonpro03) [inline Obsidian.API/Noise/TerrainSelect.cs:14]: This is a huge problem I need some help solving someday. Basically, the only way I could figure out how to merge biome's terrain together was to have them meet at sea level... which works great for overworld biomes, but looks awful when there are 2 adjacent ocean biomes.

Someday...

File: Obsidian.API/Noise/VoronoiBiomes.cs [hunk 2]
```diff
@@ -48,12 +49,12 @@ public VoronoiBiomes() : base(0)
 
     }
 
+    [SkipLocalsInit]
```
Comment (Jonpro03) [inline Obsidian.API/Noise/VoronoiBiomes.cs:52]: Neat

File: Obsidian.API/Noise/VoronoiBiomes.cs [hunk 3]
```diff
@@ -79,22 +81,50 @@ public override double GetValue(double x, double y, double z)
...
 
+        var meVal = NoiseGenerator.ValueNoise3D((int)Math.Floor(me.Point.x), 0, (int)Math.Floor(me.Point.z));
+        (me.BaseBiome, me.Variant) = GetBaseBiome(meVal);
+        var nearestVal = NoiseGenerator.ValueNoise3D((int)Math.Floor(nearest.Point.x), 0, (int)Math.Floor(nearest.Point.z));
+        (nearest.BaseBiome, nearest.Variant) = GetBaseBiome(nearestVal);
+
+        return (double)ProcessBiomeRules(me, nearest);
+    }
+
+    private static void GetMin(ReadOnlySpan<VoronoiCell> cells, ref VoronoiCell min, ref VoronoiCell secondMin)
```
Comment (Jonpro03) [inline Obsidian.API/Noise/VoronoiBiomes.cs:100]: Curious. I don't understand why this is more optimized than `Sort()`

**Review by Jonpro03 — APPROVED** [review]
LGTM

File: Obsidian.API/Noise/VoronoiBiomes.cs (hunk 3 above)
Comment (Seb-stian) [reply Obsidian.API/Noise/VoronoiBiomes.cs:100]: `O(n log n)` → `O(n)`

**Pull request merged**